src/loaders/lottie/tvgLottieBuilder.cpp

```diff
@@ -1053,12 +1053,21 @@ void LottieBuilder::updateText(LottieLayer* layer, float frameNo)
             }
         }
 
+        auto pp = *p;
```

Comment:
updateText() is horrible now. At least, it is necessary to add appropriate comments in the right places so that readers can understand the brief context flow.

Reply:
I'd rather avoid refactoring updateText in this pr if possible. but I know it's has to be done at some point. added comments